File: advJava/20-sep-jpa-demo/src/main/java/com/empower/ecom/service/ProductService.java
package com.empower.ecom.service;

import java.util.List;
import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.empower.ecom.model.Product;
import com.empower.ecom.model.ProductRepository;

@Service
public class ProductService {
	@Autowired
	private ProductRepository pr;
	
	public Product addProduct(Product product) {
		return pr.save(product);
	}
	public List<Product> retrieveAllProducts() {
		return pr.findAll();
	}
	public Product findProductById(Integer id) {
		Optional<Product> temp = pr.findById(id);
		Product product=null;
		if(temp.isPresent())
		{
			product=temp.get();
		}
		return product;
	}
	public Product updateProduct(Product product) {
		//if product is already present then only update
		Product p = findProductById(product.getId());
		if(p!=null)
		{
			p=product;
			pr.save(p);
		}
		return p;
	}
	public Product deleteProduct(Integer id) {
		Product p = findProductById(id);
		if(p!=null)
		{
			pr.delete(p);
		}
		return p;
	}
}

File: React/product/src/Services/ProductService.js
import axios from "axios";

class ProductService
{
    URL='http://localhost:5000/product'

    getAllProducts()
    {
        return axios.get(this.URL);
    }

    findProductById(id)
    {
        return axios.get(this.URL+"/"+id);
    }

    addProduct(product)
    {
        return axios.post(this.URL,product);
    }

    updateProduct(product)
    {
        return axios.put(this.URL,product);
    }

    deleteProduct(id)
    {
        return axios.delete(this.URL+"/"+id);
    }
}
export default new ProductService();